### DISCUSSION TITLE
How does the package handle time series resampling?

### DISCUSSION THREAD

**Discussion by amywinder900** [neuralhydrology/neuralhydrology#164] [Q&A] [answered]
I'm trying to get my head around what's going on in the code when the data is resampled.

In my situation, my input data has a 15 minute resolution, and my model is using this at a daily and hourly resolution. Clearly some processing is happening to make this change, but I can't understand where it's happening. 

My dataset includes flow m3/s and rainfall data mm, which may require different handling. For example resampling the flow by taking the flow at the same time every day would be appropriate, but doing the same for rainfall would make that input meaningless.

**Comment by gauchm** [comment, marked as answer]
The resampling happens [here](https://github.com/neuralhydrology/neuralhydrology/blob/master/neuralhydrology/datasetzoo/basedataset.py#L554). Regardless of the variable, the code will do a mean.

If you want to do something other than mean, I suggest to do this as a preprocessing step before running training/evaluation.

That said, without knowing your application, maybe it's also worth thinking about whether it's actually necessary to aggregate different in-/outputs differently. E.g., a typical LSTM won't care whether it gets `mean(rain_0, ..., rain_23)` or `sum(rain_0, ..., rain_23)` --- the difference is just a constant factor of 24 (which might be removed by scaling anyways).

**Reply by amywinder900** [reply]
Good point, thank you very much
